Remove CircleCI

jkmassel opened the pull request:
Removes CircleCI – replacing its last use (release builds) with API calls to Buildkite.

jkmassel committed:
Remove CircleCI

jkmassel committed:
Remove CircleCI Token from user.env-example

jkmassel committed:
Remove deprecated lane

jkmassel committed:
Fix keychain issues

jkmassel commented:
<!-- REUSE_ID: jetpack-installable-build-link -->You can test the <strong>Jetpack</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/jetpack-installable-builds/'>here</a> with build number: <code>pr17891-f1728dd</code>. IPA is available <a href='https://appcenter-filemanagement-distrib1ede6f06e.azureedge.net/26759510-daeb-419a-b7a5-16f660eccbe9/Jetpack%20Alpha.ipa?sv=2019-02-02&sr=c&sig=jPAE8jfLx3uSMDt1tuO0b%2FQ0aCdQh8SDt7GfdGRZUGk%3D&se=2022-02-10T21%3A51%3A11Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.

jkmassel commented:
<!-- REUSE_ID: installable-build-link -->You can test the <strong>WordPress</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/WPiOS-One-Offs/'>here</a> with build number: <code>pr17891-f1728dd</code>. IPA is available <a href='https://appcenter-filemanagement-distrib4ede6f06e.azureedge.net/2730ec48-804a-4b0a-96ed-5b8349f719d6/WordPress%20Alpha.ipa?sv=2019-02-02&sr=c&sig=32yqe3tsyssoFT4%2BA1B9OgBLY55wpbqRcdinQcu5GZo%3D&se=2022-02-10T21%3A52%3A45Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.

jkmassel commented:
@mokagio – this should be the last piece of the puzzle – once this lands you should have all the Buildkite tooling as part of the next release!

mokagio commented on README.md:
```diff
@@ -1,6 +1,6 @@
 # WordPress for iOS #
 
-[![CircleCI](https://circleci.com/gh/wordpress-mobile/WordPress-iOS.svg?style=svg)](https://circleci.com/gh/wordpress-mobile/WordPress-iOS)
+[![Build status](https://badge.buildkite.com/2f3fbb17bfbb5bba508efd80f1ea8d640db5ca2465a516a457.svg)](https://buildkite.com/automattic/wordpress-ios)
```
We could also remove this wholesale, given GitHub reports the build status for each commit these days, something that didn't happen at the time the badges became trendy.

mokagio reviewed (changes requested):
Apologies @jkmassel, I just noticed I missed something when reviewing #17862. 

I could be wrong, but it looks to me like there is no way currently to tell the `release-builds.yml` pipeline whether it should build a beta or a stable build. In practice, I'm pretty sure the only difference is in the GitHub release always being a stable one (i.e. not having the "pre-release" checkbox selected) but I think it's something we ought to address.

I know you're AFK today and this work affect me directly, so I'm going to:

- Request changes to avoid this being merged accidentally
- Open a PR against this branch to address this

---

Update: Also, this doesn't use a version of release toolkit that includes the `buildkite_trigger_build` action, yet.

mokagio committed:
Point release toolkit to `trunk` to get `buildkite_trigger_build` action

mokagio committed:
Remove `build_and_upload_beta_release` lane, recently become unused

mokagio committed:
Remove `build_and_upload_stable_release` lane, recently become unused

mokagio committed:
DRY calls to Buildkite action in `Fastfile`

This will also make the addition of a parameter to differentiate between
beta and stable builds cleaner.

mokagio committed:
Move each parameter in `release-build-wordpress.sh` to dedicate line

We'll add another one next, and the line length would have gotten too
long

mokagio committed:
Allow `release-build-wordpress.sh` to specify whether to build a beta

mokagio committed:
Forward beta release parameter from Fastlane to Buildkite

mokagio committed:
Make `release-build-*` Buildkite scripts executable

After adding a parameter to the `release-build-wordpress.sh` call, I got
a permission denied error. See
https://buildkite.com/automattic/wordpress-ios/builds/4907#226a7e18-7592-4e95-9064-2829b4cb7f12/381-383

I guess if the path to a script is the only value in the `command` node,
then Buildkite calls it via `sh` (or maybe `$SHELL`), otherwise it runs
it as an actual command within its shell, in which case if the script is
not executable, it fails.

mokagio commented:
> Open a PR against this branch to address this

https://github.com/wordpress-mobile/WordPress-iOS/pull/17893

jkmassel committed:
Merge pull request #17893 from wordpress-mobile/remove/circle-ci-and-fix-beta-builds

Allow for discriminating between beta and stable when building releases on Buildkite

mokagio committed:
Update release toolkit to version 3.0.0 to get `buildkite_trigger_build`

mokagio committed:
Merge remote-tracking branch 'origin/release/19.2' into remove/circle-ci

Got a conflict in `Gemfile.lock`:

```diff
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@@ -161,12 -159,11 +161,18 @@@ GE
        xcodeproj (>= 1.13.0, < 2.0.0)
        xcpretty (~> 0.3.0)
        xcpretty-travis-formatter (>= 0.0.3)
++<<<<<<< HEAD
 +    fastlane-plugin-appcenter (1.11.0)
 +    fastlane-plugin-sentry (1.8.1)
 +    fastlane-plugin-wpmreleasetoolkit (3.0.0)
++=======
+     fastlane-plugin-appcenter (1.11.1)
+     fastlane-plugin-sentry (1.11.0)
+     fastlane-plugin-wpmreleasetoolkit (2.3.0)
++>>>>>>> origin/release/19.2
        activesupport (~> 5)
        bigdecimal (~> 1.4)
 +      buildkit (~> 1.4)
        chroma (= 0.2.0)
        diffy (~> 3.3)
        git (~> 1.3)```
```

The reason is that `release/19.2` updated Fastlane, which in turn
updated the App Center and Sentry plugins, while this branch updated the
release toolkit.

I solved the conflict by keeping both sets of updates. I then run
`bundle install` to verify the `Gemfile.lock` was functional.

mokagio reviewed (approved):
Now that both the fix to differentiate between beta and stable and the release toolkit update have been implemented, we can ship this!

Looking forward test this in action as soon as there'll be a new 19.2 beta to ship 🚀

mokagio commented:
Ouch!

Auto-merge kicked in because there are no GitHub checks requirements on the release branch 🤔 

<img width="962" alt="image" src="https://user-images.githubusercontent.com/1218433/153295139-2ee2ea7c-4ae6-46f0-9eb4-581641004bc2.png">

mokagio commented:
It looks like Octokit exposes a way to set required checks while setting the branch protection: https://github.com/octokit/octokit.rb/blob/35d968fddf43f8cd7618e082e17e958be67964f1/spec/octokit/client/repositories_spec.rb#L225-L228

Maybe we should look into updating the release toolkit to use that? 🤔

jkmassel commented:
> Maybe we should look into updating the release toolkit to use that? 🤔

Agreed!!

AliSoftware commented:
@mokagio that's why I opened pdnsEh-7d-p2 :D Maybe it would be worth bumping the priority of that one a bit, especially given it probably won't take _that much_ time to implement? (famous last words…)
